src/Microsoft.SqlTools.Hosting/Hosting/Protocol/RequestContext.cs

```diff
@@ -37,13 +38,22 @@ public RequestContext(Message requestMessage, MessageWriter messageWriter)
                 eventParams);
         }
 
-        public virtual async Task SendError(object errorDetails)
+        public virtual async Task SendError(string errorMessage, int errorCode = 0, object data = null)
```

Comment:
>public virtual async Task SendError(string errorMessage, int errorCode = 0, object data = null) [](start = 8, length = 95)

Can we add an overload that takes an exception? Seems like a common use case. This overload would:
1) Set the message to exception.Message
2) Set the data to the exception.ToString()
3) Maybe, if a SqlException or Win32Exception, set the error code